packages/rust/server/src/services/email.rs

```diff
@@ -1,10 +1,94 @@
+use crate::config::CONFIG;
+use sendgrid::SGClient;
+use sendgrid::SendgridError;
+use sendgrid::{Destination, Mail};
+
+#[derive(Default, Clone, Copy)]
+pub struct SendGridEmailService {}
 #[derive(Default, Clone, Copy)]
-pub struct EmailService {}
+pub struct PrinterEmailService {}
+
+pub trait EmailService {
+    fn send(
+        &self,
+        email: &str,
+        subject: &str,
+        recipient: &Vec<(String, String)>,
```

Comment:
what's the difference between email and recipient? and do we need a Vec or will a slice work?

Reply:
This is an array of tuples/pairs that package a recipient's name and email address. I could make it a slice.